Add Geocortex Reporting section (#76)
geocortex/dev-center: docs/reporting/key-concepts.mdx

---
title: Key Concepts
description: Geocortex Reporting - Learn about key concepts of Geocortex Reporting

> hide_table_of_contents: true

Reviewer: We have occasionally disabled the TOC when it didn't make sense for the page. In this case i think we can remove this option as the headers make sense IMO.

## Geocortex Report Designer

Geocortex Report Designer provides an extensive suite of tools for laying out reports, specifying the data that you want to include, and presenting the data using charts, tables, and a wide variety of other configurable elements. Organizations can sign into the cloud-hosted report designer, or they can host it themselves on their own web server. They can then author as many report templates as their application requirements demand and save them into ArcGIS Online or Portal for ArcGIS as items. Sharing reports with end users allows them to create on-demand, print-ready PDF reports about maps and features, using data from the user's current session.

Report Designer has a built-in Preview environment for running reports. Using the Preview is the quickest way to test a report while you are designing it.

## Geocortex Reporting Service

The [Geocortex Reporting REST API](https://apps.geocortex.com/reporting/service/api-viewer/index.html?url=../swagger.json#/) can be used to asynchronously run a report, watch for report completion and download the completed report as a PDF.

> The basic steps for running a report are:

Reviewer 2: Do we need to add a prerequisite that says that you must have already created a report?

Author: Added a note about using Designer to create a report first.

> 1. Obtain a run token from the [/auth/token/run](https://apps.geocortex.com/reporting/service/api-viewer/index.html?url=../swagger.json#/auth/post_auth_token_run) endpoint.

Reviewer 2: Should we add a sub-bullet to say this is only required if the report is secured or if the reports access secured content?

Author: Done

> 2. Run the report with a request to [/job/run](https://apps.geocortex.com/reporting/service/api-viewer/index.html?url=../swagger.json#/job/post_job_run).

Reviewer 2: Should we add sub-bullets that say that the request body will specify the ID of the report to run as well as any parameters to pass to it?

Author: I had intended for this to be very high-level and plan to include further details and sample header/body content in the API section.

Author: I'll go ahead and add some bullets, I can always remove them later once the API section is filled in.

Author: I had intended for this to be very high-level and plan to include further details and sample header/body content in the API section.

Author: I'll go ahead and add some additional info, we can always remove it later.

Author: Added example body and header content.

Reviewer 2: The info looks good here. I think we just need to deal with the back ticks mentioned in Ian's comment.

Reviewer 2: One other thing. The example JWT, doesn't load... and that is a good thing. I just want to confirm it is sufficiently mangled/made up so that there is no way to recover anything meaningful from it.

Author: I did mangle the JWT, but I've now truncated it to eliminate the horizontal scroll bar.

---
title: Key Concepts
description: Geocortex Reporting - Learn about key concepts of Geocortex Reporting
---

import useBaseUrl from "@docusaurus/useBaseUrl";
import DesignerCallout from "./snippets/designer-callout.mdx";

<DesignerCallout />

## Reports

A report is a formatted presentation of a particular type of information. Geocortex Reporting specializes in presenting data and images related to web maps.

> Reports are designed by report authors (administrators) using a web-based tool called Geocortex Report Designer. They are saved to the Geocortex Reporting service, with an item in the content area of the author's ArcGIS account. To allow users to run a report, the author shares the ArcGIS item with them. End users use the reports designed by the author to create on-demand, print-ready reports about maps and their features.

Reviewer: ```suggestion
Reports are designed by report authors (administrators) using a web-based tool called Geocortex Report Designer. They are saved to the Geocortex Reporting service, along with an item in the content area of the author's ArcGIS account. To allow users to run a report, the author shares the ArcGIS item with them. End users use the reports designed by the author to create on-demand, print-ready reports about maps and their features.
```

End users run reports from web applications that the author has configured to host the reports (host applications). Alternatively, a report can be run using the Geocortex Reporting REST API. The output report is a PDF.

## Geocortex Report Designer

[Geocortex Report Designer](https://apps.geocortex.com/reporting/designer/) provides an extensive suite of tools for laying out reports, specifying the data that you want to include, and presenting the data using charts, tables, and a wide variety of other configurable elements. Organizations can sign into the cloud-hosted report designer, or they can host it themselves on their own web server. They can then author as many report templates as their application requirements demand and save them into ArcGIS Online or Portal for ArcGIS as items. Sharing reports with end users allows them to create on-demand, print-ready PDF reports about maps and features, using data from the user's current session.

> Report Designer has a built-in Preview environment for running reports. Using the Preview is the quickest way to test a report while you are designing it.

Reviewer: ```suggestion
Report Designer has a built-in preview environment for running reports. Using the preview is the quickest way to test a report while you are designing it.
```

## Geocortex Reporting Service

The [Geocortex Reporting REST API](https://apps.geocortex.com/reporting/service/api-viewer/index.html?url=../swagger.json#/) can be used to asynchronously run a report, watch for report completion and download the completed report as a PDF.

:::note
Prior to running a report, you must create a new report template using [Geocortex Report Designer](https://apps.geocortex.com/reporting/designer/).
:::

The basic steps for running a report are:

1.  Obtain a run token from the [/auth/token/run](https://apps.geocortex.com/reporting/service/api-viewer/index.html?url=../swagger.json#/auth/post_auth_token_run) endpoint.
    -   Only required if the report is secured or accesses secured content
2.  Run the report with a request to [/job/run](https://apps.geocortex.com/reporting/service/api-viewer/index.html?url=../swagger.json#/job/post_job_run).

>     -   The request body requires a template object which consists of an ArcGIS Portal url and itemid. An optional array of parameters may also be passed to the report.

Reviewer: ```suggestion
    -   The request body requires a template object which consists of a `portalUrl` and `itemid`. An optional array of `parameters` may also be passed to the report.
```

    <!-- cspell:disable -->

>         ```json

Reviewer: Something is up with these backticks on this page. Perhaps the indentation is messing with it? I see it on the page: https://deploy-preview-76--geocortex-dev-center.netlify.app/docs/reporting/key-concepts

Author: Fixed

        {
            "template": {
                "itemid": "5a810f3ac13454d6bfd2ac22ed0cc4b7",
                "portalUrl": "https://arcgis.com"
            },

>             "parameters": [...]

Reviewer: Maybe an example would be helpful here? Something like (will likely need prettier formatting):

```suggestion
            "parameters": [
            	{
                    "name": "SomeParam",
                    "containsMultipleValues": true,
                    "values": [1, 2]
                },
                {
                    "name": "AnotherParam", 
                    "containsMultipleValues": false, 
                    "value": 42
                }
            ]
```
